app/src/main/java/com/example/android/architecture/blueprints/todoapp/data/source/DefaultTasksRepository.kt

```diff
@@ -110,9 +106,7 @@ class DefaultTasksRepository(
 
     override suspend fun completeTask(taskId: String) {
         withContext(ioDispatcher) {
-            (getTaskWithId(taskId) as? Success)?.let { it ->
-                completeTask(it.data)
-            }
+            getTaskWithId(taskId)?.run { completeTask(this) }
```

Comment:
Nit: `apply` or `let` instead of `run`? You're modifying the task, it's not a random side-effect

Reply:
Unfortunately `also` doesn't work because there are 2 `completeTask` methods. Will keep it simple and change to `let`.